libs/pageserver_api/src/models.rs

```diff
@@ -1328,7 +1331,19 @@ impl PagestreamBeMessage {
 
             Self::GetPage(resp) => {
                 bytes.put_u8(Tag::GetPage as u8);
-                bytes.put(&resp.page[..]);
+                match protocol_version {
+                    PagestreamProtocolVersion::V1 | PagestreamProtocolVersion::V2 => {
+                        bytes.put(&resp.page[..])
+                    }
+                    _ => {
```

Comment:
nit: again, let's be explicit about the match (`PagestreamProtocolVersion::V3`). Makes it easier to implement V4 without mistakes.